### PR TITLE
feat: way to avoid github API limit (add tests + docs)

### PR DESCRIPTION
**Pull request by jakebolam**
Follow-up to https://github.com/jfmengels/all-contributors-cli/pull/122, adding tests and docs. Also converted the pattern to follow what was done for gitlab.

**What**:
Add a way to avoid GITHUB API LIMIT by supplying an optional GITHUB_AUTH token to increase github rate limiting from 50 requests to 5000 (and limited by user, not IP)

**Why**:
Fixes: #121 

**How**:
By Me

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,20 +210,18 @@ These are the keys you can specify:
...
...its?author=kentcdodds "Code") | [<img src="https://avatars.githubusercontent.com/u/14871650?v=3" width="100px;"/><br /><sub><b>João Guimarães</b></sub>](https://github.com/jccguimaraes)<br />[💻](https://github.com/jfmengels/all-contributors-cli/commits?author=jccguimaraes "Code") | [<img src="https://avatars.githubusercontent.com/u/1282980?v=3" width="100px;"/><br /><sub><b>Ben Briggs</b></sub>](http://beneb.info)<br />[💻](https://github.com/jfmengels/all-contributors-cli/commits?author=ben-eb "Code") | [<img src="https://avatars.githubusercontent.com/u/22768990?v=3" width="100px;"/><br /><sub><b>Itai Steinherz</b></sub>](https://github.com/itaisteinherz)<br />[📖](https://github.com/jfmengels/all-contributors-cli/commits?author=itaisteinherz "Documentation") [💻](https://github.com/jfmengels/all-contributors-cli/commits?author=itaisteinherz "Code") | [<img src="https://avatars.githubusercontent.com/u/5701162?v=3" width="100px;"/><br /><sub><b>Alex Jover</b></sub>](https://github.com/alexjoverm)<br />[💻](https://github.com/jfmengels/all-contributors-cli/commits?author=alexjoverm "Code") [📖](https://github.com/jfmengels/all-contributors-cli/commits?author=alexjoverm "Documentation") |
```
Comment (jakebolam) [inline README.md]: it appears running `yarn kcd-scripts contributors generate` does not work locally 🤔

File: .all-contributorsrc [hunk 2]
```diff
@@ -232,7 +232,7 @@
       "avatar_url": "https://avatars2.githubusercontent.com/u/8073251?v=4",
       "profile": "https://github.com/xuchaoying",
       "contributions": [
-        "code",
```
Comment (jakebolam) [inline .all-contributorsrc]: This was causing all-contributors to crash hard with a useless error message `Cannot read property 'then' of null` I'm going to propose we improve the parsing of the config file, to alert on issues like this

**Comment by jakebolam** [comment]
@kentcdodds added tests/docs and fixed all-contributors (was previously broken with an invalid json files). Follow up to #122

File: README.md (hunk 1 above)
Comment (jakebolam) [reply README.md]: Turns out `.all-contributorsrc` JSON file was invalid and was crashing silently on generate

Comment (jakebolam) [reply README.md]: Spent an hour debugging in node trying to find this. Have added better support with this in PR: https://github.com/jfmengels/all-contributors-cli/pull/124/files

File: .all-contributorsrc (hunk 2 above)
Comment (jakebolam) [reply .all-contributorsrc]: added better detection of this in: https://github.com/jfmengels/all-contributors-cli/pull/124/files

File: README.md [hunk 3]
```diff
@@ -210,6 +210,10 @@ These are the keys you can specify:
 }
 ```
 
+In some cases you may see the error message 'GitHub API rate limit exceeded for xxx'. You may need to set an environment variable named `PRIVATE_TOKEN` in order to circumvent this [GitHub rate limit](https://developer.github.com/v3/rate_limit/).
```
[the remark this answers is not in the document]
Comment (jakebolam) [reply README.md]: I tied in to the same way this project supports gitlab: https://github.com/jfmengels/all-contributors-cli/blob/master/src/repo/index.js#L4

[the remark this answers is not in the document]
Comment (kentcdodds) [reply README.md]: Gotcha. Thanks!

**Comment by kentcdodds** [comment]
I can merge this, but the build is broken and we can't have a release until it's fixed (the release is automated). I don't have time to look into it I'm afraid. If you'd like to investigate and open a PR for that, I'd appreciate it. Thanks!

**Pull request merged**